sklearn/base.py

```diff
@@ -435,6 +437,22 @@ def _validate_data(self, X, y=None, reset=True,
 
         return out
 
+    def _repr_html_(self):
+        """HTML or string representation of an estimator depending on
+        global configuration flag `repr_html`.
+
+        Read more in the :ref:`User Guide <visualizing_composite_estimators>`.
+
+        Returns
+        -------
+        repr: str
+            HTML or string representation of estimator.
+        """
+        repr_html = get_config()["repr_html"]
+        if repr_html:
+            return estimator_repr_html(self)
+        return repr(self)
```

Comment:
```suggestion
        return f"<code>{html.escape(repr(self))}</code>"
```
maybe?

Reply:
That would still break people checking the output in jupyter notebooks in their CI, right?

Reply:
Yes, it would. What does ipython.display do if we return None. Does it back off to another repr?

Reply:
Core Jupyter developer here: I think you should use _repr_mimebundle_ instead of shoving the text repr into the html repr as per my comment above.